modules/nf-core/dorothea/differential/environment.yml

```diff
@@ -0,0 +1,12 @@
+name: dorothea_differential_limma
+
+channels:
+  - conda-forge
+  - bioconda
+  - defaults
+dependencies:
+  - bioconda::bioconductor-decoupler=2.8.0
```

Comment:
You shouldn't need to do this- it should just be a case of depending on [bioconductor-dorothea](https://anaconda.org/bioconda/bioconductor-dorothea) and the associated biocontainers. See other R modules such as deseq2/differential.

Reply:
Got it. Will change it.

Reply:
They come from the decoupleR package which uses CollecTRI (the updated Dorothea basically). It does produce the plots requested.

Reply:
I think what you actually need to make is a simple module wrapper around `decoupler/run_ulm`. Dorothea is just the data source here, right?

`decoupler/run_ulm` would take an additional input of the network, so that it could function generically.

I'd also discourage inclusion of extensive additional code (e.g. your plots), since that code then has to be maintained by someone within the module. Better to keep a thin wrapper around `decoupler/run_ulm`.

You could write an additional module under `custom` to do your plotting- that way a user is clear what code (and bugs!) come from the developers of decoupler, and which from your custom code.

Reply:
Makes sense. So just to recap, I would :
- remove the plotting 
- Add additional input for the network (Dorothea, CollecTRI, custom). 
-Optionally make an additional process for plotting 

Did I miss anything?